src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -0,0 +1,133 @@
+            return $this->respond(new ActionPayload(400, null, $error));
+        } catch (\Stripe\Exception\SignatureVerificationException $e) {
+            $logMessage = 'Invalid Signature';
+            $this->logger->warning($logMessage);
+            $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+            return $this->respond(new ActionPayload(400, null, $error));
+        }
+
+        /** @var Donation $donation */
+        $donation = $this->donationRepository->findOneBy(['transactionId' => $event->data->object->id]);
+
+        if ($event instanceof StripeEvent && $donation) {
+            // Handle the event
+            switch ($event->type) {
+                case 'payment_intent.succeeded':
+                    $this->handlePaymentIntentSucceeded($event, $donation);
+                    break;
+                case 'payment_intent.created':
+                    $logMessage = 'Unsupported Action';
+                    $this->logger->warning($logMessage);
+                    $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+                    return $this->respond(new ActionPayload(400, null, $error));
+                default:
+                    $logMessage = 'Unsupported Action';
+                    $this->logger->warning($logMessage);
+                    $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+                    return $this->respond(new ActionPayload(400, null, $error));
+            }
+        } else {
+            $logMessage = 'No Content';
+            $this->logger->warning($logMessage);
+            $error = new ActionError(ActionError::BAD_REQUEST, null ?? $logMessage);
+            return $this->respond(new ActionPayload(204, null, $error));
+        }
+
+        return $this->respondWithData($event->data->object);
+    }
+
+    public function handlePaymentIntentSucceeded(StripeEvent $event, $donation): Response
+    {
+        $missingRequiredField = (empty($event->status) ||
+            empty($event->data->object->billing_details->address->postal_code) ||
+            empty($event->data->object->billing_details->address->country) ||
+            empty($event->data->object->billing_details->email) ||
+            empty($event->data->object->billing_details->name) ||
+            !isset($event->data->metadata->coreDonationGiftAid, $event->data->metadata->optInTbgEmail) ||
+            empty($event->data->object->id));
+        if ($missingRequiredField) {
+            $message = 'Hook missing required values';
+            $this->logger->warning("Donation ID {$event->data->object->id}: {$message}");
+            $error = new ActionError(ActionError::BAD_REQUEST, $message);
+
+            return $this->respond(new ActionPayload(400, null, $error));
+        }
+
+        $donation->setDonationStatus($event->status);
+        $donation->setDonorBillingAddress($event->data->object->billing_details->address->postal_code);
```

Comment:
I'm not sure we need to set all these values that are immutable. They go to Stripe for reference so I think it's needless overhead to re-set them here. It's probably good to check they exist as expected still but we only really need to set status unless I'm missing something.